thali/thalireplicationmanager.js

```diff
@@ -192,11 +202,44 @@ function muxServerBridge(tcpEndpointServerPort) {
   var serverPlex = multiplex({}, function(stream, id) {
     var clientSocket = net.createConnection({port: tcpEndpointServerPort});
     stream.pipe(clientSocket).pipe(stream);
+
+    clientSocket.on('end', function () {
+      clientSocket.destroy();
+    });
+
+    clientSocket.on('error', function (err) {
+      console.log('incoming client sockiet error %s', err);
+      clientSocket.destroy();
+    });
   });
 
-  return net.createServer(function(incomingClientSocket) {
+  var server = net.createServer(function(incomingClientSocket) {
+    incomingClientSocket.on('end', function () {
+      console.log('incoming client socket end');
```

Comment:
Why do you destroy the incomingClientSocket on error but not on end?  Of course given my previous question about destroying "end"ed sockets maybe this is correct and line 207 is the problem?